File: dodam-core/src/main/java/b1nd/dodamcore/member/repository/MemberRepository.java
package b1nd.dodamcore.member.repository;

import b1nd.dodamcore.member.domain.entity.Member;
import b1nd.dodamcore.member.domain.enums.AuthStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Repository
public interface MemberRepository extends JpaRepository<Member, String> {

    Optional<Member> findById(String id);

    Optional<Member> findByIdAndPw(String id, String pw);

    List<Member> findByJoinDateAfter(LocalDateTime joinDate);

    List<Member> findByStatus(AuthStatus status);

    //Optional<Member> findByStudent(Student student);
}

File: services/service-out-sleeping/src/main/kotlin/com/b1nd/dodamdodam/outsleeping/domain/member/repository/MemberRepository.kt
package com.b1nd.dodamdodam.outsleeping.domain.member.repository

import com.b1nd.dodamdodam.outsleeping.domain.member.entity.MemberEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface MemberRepository : JpaRepository<MemberEntity, Long> {

    fun findByUserId(userId: UUID): MemberEntity?

    fun findAllByUserIdIn(userIds: Collection<UUID>): List<MemberEntity>

    fun findAllByRole(role: String): List<MemberEntity>
}
